sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/FileTypeSensorTest.java

```diff
@@ -0,0 +1,188 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.dotnet.shared.plugins;
+
+import java.io.File;
+import java.nio.file.Path;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.ArgumentCaptor;
+import org.sonar.api.batch.fs.InputFile.Type;
+import org.sonar.api.batch.fs.internal.DefaultInputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.config.Settings;
+import org.sonar.api.utils.log.LogTester;
+import org.sonar.api.utils.log.LoggerLevel;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+public class FileTypeSensorTest {
+  public static final String LANG_KEY = "LANG_KEY";
+
+  @Rule
+  public LogTester logTester = new LogTester();
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  private Path workDir;
+
+  private Settings settingsMock;
+  private SensorContextTester tester;
+  private ProjectTypeCollector projectTypeCollectorMock;
+  private DotNetPluginMetadata pluginMetadataMock;
+  private FileTypeSensor sensor;
+
+  @Before
+  public void prepare() throws Exception {
+    workDir = temp.newFolder().toPath();
+    settingsMock = mock(Settings.class);
+    tester = SensorContextTester.create(new File("src/test/resources"));
+    tester.fileSystem().setWorkDir(workDir);
+    tester.setSettings(settingsMock);
+
+    projectTypeCollectorMock = mock(ProjectTypeCollector.class);
+
+    pluginMetadataMock = mock(DotNetPluginMetadata.class);
+    when(pluginMetadataMock.languageKey()).thenReturn(LANG_KEY);
+
+    sensor = new FileTypeSensor(projectTypeCollectorMock, pluginMetadataMock);
+  }
+
+
+  @Test
+  public void should_describe() {
+
+    SensorDescriptor desc = mock(SensorDescriptor.class);
+    sensor.describe(desc);
+
+    verify(desc).name("Verify what types of files (MAIN, TEST) are in LANG_KEY projects.");
```

Comment:
..and also verify that language filter is empty?

Reply:
This one seems missed. 
I'd like to add check to verify that the description language filter is empty.

Reply:
sorry, I didn't find a way to do it yesterday, I'll check again today

Reply:
Should be easy like this, right? https://github.com/SonarSource/sonar-dotnet/blob/1bf06b4b6c5aa030a31c90c27e4b2c00a95e9de7/sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/DotNetSensorTest.java#L93-L99

Will you update this in the next PR with sorting?

Reply:
I already updated using the mock, I'm not sure there's any benefit in using your suggested approach.

The way I did it ensures that it only calls what we expect it to call - setting the name.